### PR TITLE
using NoahMP tiled tq2m with the bug fix

### PR DESCRIPTION
**Pull request by HelinWei-NOAA**

### CODE REVIEW

File: physics/module_sf_noahmplsm.f90 [hunk 1]
```diff
@@ -4756,6 +4762,20 @@ subroutine bare_flux (parameters,nsnow   ,nsoil   ,isnow   ,dt      ,sag     , &
        if (parameters%urban_flag) q2b = qsfc
      end if
 
+! opt_sfc 3: fh2 is the stability
+     if(opt_sfc == 3 ) then
+       ehb2  = fv*vkc/fh2
```
Comment (barlage) [inline physics/module_sf_noahmplsm.f90:4767]: do we need any check for fh2 close to zero here?

File: physics/module_sf_noahmplsm.f90 [hunk 2]
```diff
@@ -4289,13 +4289,19 @@ subroutine vege_flux(parameters,nsnow   ,nsoil   ,isnow   ,vegtyp  ,veg     , &
...
+   if (opt_sfc == 1 .or. opt_sfc == 2 ) then
 !      cah2 = fv*1./vkc*log((2.+z0h)/z0h)
       cah2 = fv*vkc/log((2.+z0h)/z0h)
       cah2 = fv*vkc/(log((2.+z0h)/z0h)-fh2)
       cq2v = cah2
    endif
 
+! opt_sfc 3: fh2 is the stability
+    if (opt_sfc ==3) then
+        cah2 = fv*vkc/fh2
```
Comment (barlage) [inline physics/module_sf_noahmplsm.f90:4301]: check for fh2~0

File: physics/module_sf_noahmp_glacier.f90 [hunk 3]
```diff
@@ -1464,6 +1464,11 @@ subroutine glacier_flux (nsoil     ,nsnow   ,emg     ,isnow   ,df      ,dzsnso
 ! 2m air temperature
      ehb2  = fv*vkc/(log((2.+z0h)/z0h)-fh2)
      cq2b  = ehb2
+! for opt_sfc 3
+     if (opt_sfc ==3) then
+       ehb2  = fv*vkc/fh2
```
Comment (barlage) [inline physics/module_sf_noahmp_glacier.f90:1469]: check for fh2~0?

**Comment by barlage** [comment]
in general, do we need a check for fh2 close to zero? or is that possible?

File: physics/module_sf_noahmp_glacier.f90 (hunk 3 above)
Comment (wzzheng90) [reply physics/module_sf_noahmp_glacier.f90:1469]: No, fh2 > 0.

Comment (barlage) [reply physics/module_sf_noahmp_glacier.f90:1469]: OK, I didn't see any limit checks in stability subroutine

Comment (wzzheng90) [reply physics/module_sf_noahmp_glacier.f90:1469]: The stability becomes zero only under some conditions which we should avoid.

**Comment by HelinWei-NOAA** [comment]
https://github.com/ufs-community/ccpp-physics/pull/25

created for HR1

**Pull request merged**